biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java

```diff
@@ -136,36 +136,57 @@ private void addClasspath(Parameters dependencies, Container c) throws IOExcepti
 		Jar jar = new Jar(file);
 		super.addClasspath(jar);
 		project.unreferencedClasspathEntries.put(jar.getName(), c);
-		if ((dependencies != null) && !Boolean.parseBoolean(c.getAttributes()
-			.getOrDefault("maven-optional", "false"))) {
-			jar.getResources(pomPropertiesFilter)
-				.forEachOrdered(r -> {
-					UTF8Properties pomProperties = new UTF8Properties();
-					try (InputStream in = r.openInputStream()) {
-						pomProperties.load(in);
-					} catch (Exception e) {
-						logger.debug("unable to read pom.properties resource {}", r, e);
-						return;
-					}
-					String depVersion = pomProperties.getProperty("version");
-					String depGroupId = pomProperties.getProperty("groupId");
-					String depArtifactId = pomProperties.getProperty("artifactId");
-					if ((depGroupId != null) && (depArtifactId != null) && (depVersion != null)) {
-						Attrs attrs = new Attrs();
-						attrs.put("groupId", depGroupId);
-						attrs.put("artifactId", depArtifactId);
-						attrs.put("version", depVersion);
-						attrs.put("scope", c.getAttributes()
-							.getOrDefault("maven-scope", getProperty(MAVEN_SCOPE, "compile")));
-						String key = new StringBuilder().append(depGroupId)
-							.append(':')
-							.append(depArtifactId)
-							.append(':')
-							.append(depVersion)
-							.toString();
-						dependencies.add(key, attrs);
-					}
-				});
+		Map<String, String> containerAttributes = c.getAttributes();
```

Comment:
This solution is a lot better! 

However, could we somehow delegate this very mavenish code to not be in the build package? Maybe a plugin? At least make all this maven support go to one place?

Nice work!

Reply:
It may be possible but we need to collect the maven dependency information at the time we make the classpath here in ProjectBuilder. Any later and we have lost the necessary information (the Containers). This is why the code generates the -maven-dependencies instruction so that the information collected here, when we have the information, can be communicated to the future when PomResource is generating the pom. By the time you get to PomResource, you have lost all information on the class path.

The -maven-dependencies instruction also allows the user to fix/replace the maven dependency information in their bnd file.

For now, I think this is fine for 5.1. We can always try and make it better in the next release.